src/build/plugins/externals.ts

```diff
@@ -301,6 +301,33 @@ export function externals(opts: NodeExternalsOptions): Plugin {
         }
       }
 
+      const userPkg = await readPackageJSON(
+        opts.rootDir || process.cwd()
+      ).catch(() => ({}) as PackageJson);
+
+      const userPkgAliases = Object.fromEntries(
+        [
+          ...Object.entries(userPkg.dependencies || {}),
+          ...Object.entries(userPkg.devDependencies || {}),
+        ]
+          .map(([name, version]) => {
+            // TODO: optimize this package name parsing
+            // { "lodash4": "npm:lodash@4.17.21" }              => [ "lodash" ~> "lodash4" ]
+            // { "@std/ulid": "npm:@jsr/std__ulid@1.0.0-rc.4" } => [ "@jsr/std__ulid" ~> "@std/ulid" ]
+
+            if (!version.includes(":")) {
+              return null;
+            }
+
+            const parts = version.split(":")[1]!.split("@");
```

Comment:
This is not a reliable way.

- Aliases can be done using `resolutions`, `overrides`, or aliases like this
- Rules can also be from workspace settings
- Package aliases can happen **at any nested level** and are respected by package managers.

Reply:
I see, then what about adding package aliases here:
https://github.com/nitrojs/nitro/blob/32e1850760fa19466ab69d31636ead05a856dc6c/src/build/plugins/externals.ts#L137-L139

If the imported package name doesn't match the resolved package name (also considering "Subpath export and full paths"), then there might be a package alias. We could automatically add this alias to `traceAlias`. How does that sound? 🤔